solver/src/solver/http_solver.rs

```diff
@@ -500,6 +522,26 @@ fn compute_fee_connected_tokens(liquidity: &[Liquidity], native_token: H160) ->
     fee_connected_tokens
 }
 
+/// Compute the scaling rate from a Balancer pool's scaling exponent.
+///
+/// This method returns an error on any arithmetic underflow when computing the
+/// token decimals. Note that in theory, this should be impossible to happen.
+/// However, we are extra careful and return an `Error` in case it does to avoid
+/// panicking. Additionally, wrapped math could have been used here, but that
+/// would create invalid settlements.
+fn compute_scaling_rate(scaling_exponent: u8) -> Result<U256> {
+    // Balancer `scaling_exponent`s are `18 - decimals`, we want the rate which
+    // is `10 ** decimals`.
+    let decimals = 18_u8.checked_sub(scaling_exponent).ok_or_else(|| {
+        anyhow!("underflow computing decimals from Balancer pool scaling exponent")
+    })?;
+
+    debug_assert!(decimals <= 18);
+    // `decimals` is guaranteed to be between 0 and 18, and 10**18 cannot
+    // cannot overflow a `U256`, so we do not need to use `checked_pow`.
+    Ok(U256::from(10).pow(decimals.into()))
+}
+
```

Comment:
I am expecting that DAI will have a "scaling_rate" of 10^18 . That would mean that DAI has a "scaling_exponent" of 1. Does that make sense?

Reply:
I believe DAI will have a scaling exponent of 0 if this was implemented as expected since scaling exponent satisfies this equation 

```18 = scaling_exponent + token_decimals```

Reply:
So the "scaling_rate" value that is passed to the http solver equals 10^18 for DAI or not?

Reply:
Oh... right yes, it appears so... DAI has `scaling_exponent` of 0 because of having 18 decimals so the value calculated for scaling_rate is 10^18.

Reply:
> I am expecting that DAI will have a "scaling_rate" of 10^18 . 

Good! So am I :smile:.

> That would mean that DAI has a "scaling_exponent" of 1. Does that make sense?

As @bh2smith said, the scaling exponent is 0. The scaling rate is calculated as:
```
scaling_rate_T = 10 ** (18 - scaling_exponent_T)

For DAI, where scaling_exponent = 0

scaling_rate_DAI = 10 ** (18 - scaling_exponent_DAI)
                 = 10 ** (18 - 0)
                 = 10 ** 18
```

Reply:
Great, thanks all!

Reply:
Just ran locally and confirmed that the scaling rates are `10 ** decimals`:
```js
      "scaling_rates":{
        "0x6b175474e89094c44da98b954eedeac495271d0f":"1000000000000000000", // DAI (18 decimals)
        "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48":"1000000", // USDC (6 decimals)
        "0xdac17f958d2ee523a2206206994597c13d831ec7":"1000000" // USDT (6 decimals)
      },
```